src/java.base/share/classes/java/io/InputStream.java

```diff
@@ -55,6 +56,32 @@ public abstract class InputStream implements Closeable {
 
     private static final int DEFAULT_BUFFER_SIZE = 8192;
 
+    /**
+     * Only used in {@link #skip(long) skip}, for avoiding multiple times byte array allocation for a same InputStream Object.
+     *
+     * Making this static would increase performance,
+     * but would make it possible to build a malicious subclass of InputStream to steal data from other InputStream Objects.
+     * So DO NOT try to make it static, for the sake of security.
+     */
+    private SoftReference<byte[]> skipBufferReference;
+
+    /**
+     * Get or create a byte array for {@link #skip(long) skip}.
+     *
+     * @param size minimum length that the skip byte array must have.
+     *             notice that this param input MUST be equal or less than {@link #MAX_SKIP_BUFFER_SIZE MAX_SKIP_BUFFER_SIZE}.
```

Comment:
The "MUST be equal" reads like something will go wrong if that condition isn't satisfied.
This method does not care about the size, except in potentially resizing if the requested size is larger.

The "notice..." statement is making a statement about code in the caller, and so doesn't belong here.

Reply:
The "notice..." statement removed.